src/routes/dataset.ts

```diff
@@ -7,12 +7,45 @@ import { fetchDataset } from '../middleware/fetch-dataset';
 import { NotFoundException } from '../exceptions/not-found.exception';
 import { logger } from '../utils/logger';
 import { DatasetListItemDTO } from '../dtos/dataset-list-item';
-import { hasError, importIdValidator } from '../validators';
+import { hasError, factTableIdValidator } from '../validators';
 import { RevisionDTO } from '../dtos/revision';
-import { FileImportDTO } from '../dtos/file-import';
+import { FactTableDto } from '../dtos/fact-table';
 
 export const dataset = Router();
 
+// Special thanks ChatGPT...  The GovUK pagination algorithm
+function generateSequenceForNumber(highlight: number, end: number): (string | number)[] {
+    const sequence: (string | number)[] = [];
+
+    // Validate input
+    if (highlight > end) {
+        throw new Error(`Highlighted number must be between 1 and ${end}.`);
+    }
+
+    // Numbers before the highlighted number
+    if (highlight - 1 > 1) {
+        sequence.push(1, '...');
+        sequence.push(highlight - 1);
+    } else {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        sequence.push(...Array.from({ length: highlight - 1 }, (_, index) => index + 1));
+    }
+
+    // Highlighted number
+    sequence.push(highlight);
+
+    // Numbers after the highlighted number
+    if (highlight + 1 < end) {
+        sequence.push(highlight + 1);
+        sequence.push('...', end);
+    } else {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        sequence.push(...Array.from({ length: end - highlight }, (_, index) => highlight + 1 + index));
+    }
+
+    return sequence;
+}
```

Comment:
Ah - so this is actually reused, would be good to extract this to utils

Reply:
Will extract and reuse in follow up PR